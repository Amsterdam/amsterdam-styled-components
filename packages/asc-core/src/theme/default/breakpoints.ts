import { Theme } from '../index'

const withTypeBreakpoint = (size: number) => (type: Theme.TypeBreakpoint) =>
  `(${type}: ${size + (type === 'min-width' ? 0.02 : 0)}px)`

const sizes = {
  mobileS: 320,
  mobileM: 375,
  mobileL: 425,
  tabletS: 540,
  tabletM: 768,
  laptop: 1024,
  laptopM: 1200,
  laptopL: 1440,
  desktop: 2560,
}

const breakpoints: Theme.BreakpointsInterface = {
  mobileS: withTypeBreakpoint(sizes.mobileS),
  mobileM: withTypeBreakpoint(sizes.mobileM),
  mobileL: withTypeBreakpoint(sizes.mobileL),
  tabletS: withTypeBreakpoint(sizes.tabletS),
  tabletM: withTypeBreakpoint(sizes.tabletM),
  laptop: withTypeBreakpoint(sizes.laptop),
  laptopM: withTypeBreakpoint(sizes.laptopM),
  laptopL: withTypeBreakpoint(sizes.laptopL),
  desktop: withTypeBreakpoint(sizes.desktop),
}

export default breakpoints
