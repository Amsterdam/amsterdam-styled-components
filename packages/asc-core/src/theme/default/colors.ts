import { Theme } from '../index'

const colors: Theme.ColorInterface = {
  primary: {
    main: '#004699',
    dark: '#00387a',
  },
  secondary: {
    main: '#ec0000',
    dark: '#bc0000',
  },
  tint: {
    level1: '#ffffff',
    level2: '#fafafa',
    level3: '#f6f6f6',
    level4: '#eeeeee',
    level5: '#cccccc',
    level6: '#999999',
    level7: '#666666',
    level8: '#000000',
  },
  bright: {
    main: '#ffffff',
  },
  error: {
    main: '#ec0000',
  },
}

export default colors
