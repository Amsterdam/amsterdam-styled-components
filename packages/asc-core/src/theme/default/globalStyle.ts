import { Theme } from '../index'

const globalStyle: Theme.GlobalStyleType = ''

export default globalStyle
