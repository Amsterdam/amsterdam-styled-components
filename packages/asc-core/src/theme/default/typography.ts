import { Theme } from '../index'
import { colors } from './index'

const defaultFontFamily = 'Avenir Next, Arial, sans-serif'

const typography: Theme.TypographyInterface = {
  fontFamily: defaultFontFamily,
  fontSize: '16px',
  h1: {
    defaultColor: colors.tint.level7,
    fontSize: '24px',
    fontWeight: 700,
    letterSpacing: 'inherit',
    lineHeight: '30px',
    marginBottom: '0',
    breakpoints: {
      tabletS: {
        defaultColor: colors.tint.level7,
        fontSize: '30px',
        fontWeight: 700,
        letterSpacing: 'inherit',
        lineHeight: '38px',
        marginBottom: '0',
      },
    },
  },
  h2: {
    defaultColor: colors.tint.level7,
    fontSize: '20px',
    fontWeight: 700,
    letterSpacing: 'inherit',
    lineHeight: '28px',
    marginBottom: '0',
    breakpoints: {
      tabletS: {
        defaultColor: colors.tint.level7,
        fontSize: '24px',
        fontWeight: 700,
        letterSpacing: 'inherit',
        lineHeight: '30px',
        marginBottom: '0',
      },
    },
  },
  h3: {
    defaultColor: colors.tint.level7,
    fontSize: '20px',
    fontWeight: 700,
    letterSpacing: 'inherit',
    lineHeight: '28px',
    marginBottom: '0',
  },
  h4: {
    defaultColor: colors.tint.level7,
    fontSize: '18px',
    fontWeight: 700,
    letterSpacing: 'inherit',
    lineHeight: '25px',
    marginBottom: '8px',
  },
  h5: {
    defaultColor: colors.tint.level7,
    fontSize: '18px',
    fontWeight: 400,
    letterSpacing: 'inherit',
    lineHeight: 1,
    marginBottom: '0',
  },
  h6: {
    defaultColor: colors.tint.level7,
    fontSize: '18px',
    fontWeight: 400,
    letterSpacing: 'inherit',
    lineHeight: 1,
    marginBottom: '0',
  },
  a: {
    defaultColor: colors.tint.level7,
    fontSize: '1rem',
    fontWeight: 'inherit',
    letterSpacing: 'inherit',
    lineHeight: 1.25,
    marginBottom: '0',
  },
  p: {
    defaultColor: colors.tint.level7,
    fontSize: '16px',
    fontWeight: 400,
    letterSpacing: 'inherit',
    lineHeight: '22px',
    marginBottom: '20px',
    breakpoints: {
      tabletS: {
        defaultColor: colors.tint.level7,
        fontSize: '16px',
        fontWeight: 400,
        letterSpacing: 'inherit',
        lineHeight: '22px',
        marginBottom: '24px',
      },
    },
  },
  em: {
    defaultColor: colors.tint.level7,
    fontSize: '16px',
    fontWeight: 400,
    letterSpacing: 'inherit',
    lineHeight: 1.25,
    marginBottom: '0',
  },
  small: {
    defaultColor: colors.tint.level7,
    fontSize: '14px',
    fontWeight: 400,
    letterSpacing: 'inherit',
    lineHeight: '18px',
    marginBottom: '0',
  },
  span: {
    defaultColor: colors.tint.level7,
    fontSize: '14px',
    fontWeight: 400,
    letterSpacing: 'inherit',
    lineHeight: '18px',
    marginBottom: '0',
  },
}

export default typography
