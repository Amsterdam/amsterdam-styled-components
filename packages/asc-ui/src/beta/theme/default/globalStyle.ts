import type { Theme } from '../../types'

const globalStyle: Theme.GlobalStyleType = ''

export default globalStyle
