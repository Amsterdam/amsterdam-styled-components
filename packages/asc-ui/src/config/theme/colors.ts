import { Theme } from '@amsterdam/asc-core'

const colors: Theme.ColorInterface = {
  primary: {
    main: '#004699',
    dark: '#00387a',
  },
  secondary: {
    main: '#ec0000',
    dark: '#bc0000',
  },
  bright: {
    main: '#ffffff',
  },
  error: {
    main: '#ec0000',
  },
}

export default colors
