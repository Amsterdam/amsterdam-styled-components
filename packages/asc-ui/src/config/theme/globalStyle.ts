import { Theme } from '@amsterdam/asc-core'

const globalStyle: Theme.GlobalStyleType = ''

export default globalStyle
