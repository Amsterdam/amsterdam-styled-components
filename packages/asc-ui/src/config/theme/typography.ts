import { Theme } from '@amsterdam/asc-core';

const typography: Theme.TypographyInterface = {
  primary: 'AvenirNextLTW01-Regular", verdana, sans-serif',
  secondary: 'AvenirNextLTW01-Regular", verdana, sans-serif',
}

export default typography
