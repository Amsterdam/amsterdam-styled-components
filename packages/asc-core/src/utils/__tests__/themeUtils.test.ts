import {
  color,
  focusStyle,
  breakpoint,
  svgFill,
  getTypographyFromTheme,
} from '../themeUtils'
import breakpoints from '../../theme/default/breakpoints'
import colors from '../../theme/default/colors'
import globalStyle from '../../theme/default/globalStyle'
import typography from '../../theme/default/typography'

describe('getColorFromTheme', () => {
  const theme = {
    breakpoints,
    globalStyle,
    colors: {
      ...colors,
      primary: {
        main: '#fff',
        dark: '#000',
      },
    },
    typography,
  }

  it('should return the requested color from theme', () => {
    expect(color('primary', 'main')({ theme })).toBe('#fff')
    expect(color('primary', 'dark')({ theme })).toBe('#000')
    expect(color('primary')({ theme })).toBe('#fff')
  })

  it('should returen the default color when the colorType is not provided', () => {
    expect(color()({ theme })).toBe('#ffffff')
  })
})

describe('getTypographyFromTheme', () => {
  it('should return the requested typography from theme', () => {
    const theme = {
      breakpoints,
      globalStyle,
      colors,
      typography: {
        ...typography,
        fontSize: '16px',
      },
    }

    expect(getTypographyFromTheme(theme, 'fontSize')).toBe('16px')
  })
})

describe('focusStyle', () => {
  const theme = {
    breakpoints,
    globalStyle,
    colors: {
      ...colors,
      support: {
        ...colors.support,
        focus: '#abcde',
      },
    },
    typography,
  }

  it('should return the focusstyle from theme', () => {
    const result = focusStyle()({ theme })
    expect(result).toContain('#abcde')
  })
})

describe('breakpoint', () => {
  const theme = {
    breakpoints,
    globalStyle,
    colors,
    typography,
  }

  it("should return undefined when the breakpoint doesn't exist", () => {
    expect(breakpoint('max-width', 'not-valid')({ theme })).toBeUndefined()
  })

  it('should return the right breakpoint', () => {
    expect(breakpoint('max-width', 'desktop')({ theme })).toEqual(
      '(max-width: 2560px)',
    )
    expect(breakpoint('min-width', 'mobileL')({ theme })).toEqual(
      '(min-width: 425px)',
    )
  })
})

describe('svgFill', () => {
  const theme = {
    breakpoints,
    globalStyle,
    colors,
    typography,
  }

  it("should return een empty string when the color doesn't exist", () => {
    expect(svgFill()({ theme })).toBe('')
  })

  it('should return the right fill color for the svg', () => {
    expect(svgFill('tint', 'level5')({ theme })).toContain('fill: #000')
  })
})
