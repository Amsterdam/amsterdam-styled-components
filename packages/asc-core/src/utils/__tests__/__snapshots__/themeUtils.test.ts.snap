// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFocusStyle should return the focusstyle from theme 1`] = `
Array [
  "
  &:focus {
    outline-color: ",
  "#abcde",
  ";
    outline-style: solid;
    outline-width: medium;
  }
",
]
`;
